pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1416,4 +1416,16 @@ self: super: {
   # cabal-fmt requires Cabal3
   cabal-fmt = super.cabal-fmt.override { Cabal = self.Cabal_3_0_0_0; };
 
+  # Several gtk2hs-provided packages at v0.13.8.0 fail to build on Darwin
+  # (Until we pick up https://github.com/gtk2hs/gtk2hs/pull/293)
+  glib = appendPatch super.glib ./patches/glib.patch;
+  gio = appendPatch super.gio ./patches/gio.patch;
+  pango = appendPatch super.pango ./patches/pango.patch;
+  # In addition to the issue above, gtk in particular needs to be told on Darwin
+  # that that the supplied gtk lib uses quartz not X11 (as recommended in
+  # https://github.com/gtk2hs/gtk2hs/issues/249). This appendConfigureFlags part
+  # should remain even after we can remove the v0.13.8.0-specific appendPatch.
+  gtk = appendConfigureFlags (appendPatch super.gtk ./patches/gtk.patch)
```

Comment:
Could you either pull these patches directly from upstream (maybe using `fetchPatch`), or try to add this `-D__attribute__(x)=` flag inline here (maybe using `appendConfigureFlags`)?

If possible, we try to avoid carrying around patches directly in nixpkgs.

Reply:
Makes sense! I can't think of a way to achieve this just via command-line flags, and there isn't a canonical patch source yet, but I've just pushed an alternative. Hopefully the let-bound helper function will be relatively short-lived.

Reply:
> a canonical patch source yet

The canonical patch source would be your PR.  Look for examples of `fetchPatch` in nixpkgs to see how this is done.

Otherwise, you might be able to use the [`optP`](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/phases.html#ghc-flag--optP%20%E2%9F%A8option%E2%9F%A9) GHC flag.

Reply:
Thanks for the guidance! I've pushed an update using fetchpatch (and re-tested on both Darwin and NixOS).

There is some commonality that could be factored out into a helper function (at which point one might make the patch darwin-specific) but I've erred on the side of keeping everything local. Unconditionally applying the patch is safe, and doesn't have a huge fan-out. But I'm not sure which is most idiomatic - opinions welcomed!